login-utils/login.c

```diff
@@ -1287,6 +1288,39 @@ static void __attribute__((__noreturn__)) usage(void)
 	exit(EXIT_SUCCESS);
 }
 
+static void load_credentials(struct login_context *cxt) {
+	char *env;
+	DIR *dir;
+	struct dirent *d;
+	struct path_cxt *pc;
+
+	env = safe_getenv("CREDENTIALS_DIRECTORY");
+        if (!env)
+                return;
+
+	pc = ul_new_path("%s", env);
+	if (!pc) {
+		syslog(LOG_ERR, _("failed to initialize path context"));
+		exit(EXIT_FAILURE);
+	}
+
+	dir = ul_path_opendir(pc, NULL);
+	if (!dir) {
+		syslog(LOG_ERR, _("failed to open credentials directory"));
+		exit(EXIT_FAILURE);
+	}
+
+	while ((d = xreaddir(dir))) {
+		char *str;
+
+		if (strcmp(d->d_name, "login.noauth") == 0) {
+			ul_path_read_string(pc, &str, d->d_name);
+			if (str && strcmp(str, "yes") == 0)
```

Comment:
We could just check for the existence of the credential here and not check its value, let me know what is preferred

Reply:
The question should be: is it possible to create a boolean-like file with negative content? For example, login.noauth=no. If possible, you have to check for the content ;-) IMHO, it's safer to check the content, or we can accept an empty file as "yes", and for non-empty, check for the content. Not sure what is normal for credentials.